tests/test_file.py

```diff
@@ -68,22 +68,39 @@ def _test_file_put(self, replica: Replica, scheme: str, test_bucket: str, upload
         bundle_uuid = str(uuid.uuid4())
         version = datetime_to_version_format(datetime.datetime.utcnow())
 
-        # should be able to do this twice (i.e., same payload, different UUIDs)
-        self.upload_file(source_url, file_uuid, bundle_uuid=bundle_uuid, version=version)
-        self.upload_file(source_url, str(uuid.uuid4()))
+        with self.subTest(f"{replica}: Created returned when uploading a file with a unique payload, and FQID"):
+            self.upload_file(source_url, file_uuid, bundle_uuid=bundle_uuid, version=version)
+
+        with self.subTest(f"{replica}: Created returned when uploading a file with same payload, and different FQID"):
+            self.upload_file(source_url, str(uuid.uuid4()))
+
+        with self.subTest(f"{replica}: OK returned when uploading a file with the same payload, UUID,  version"):
+            self.upload_file(source_url, file_uuid, bundle_uuid=bundle_uuid,
+                             version=version, expected_code=requests.codes.ok)
+
+        with self.subTest(f"{replica}: OK returned when uploading a file with a different payload and same FQID"):
+            self.upload_file(source_url, file_uuid, version=version, expected_code=requests.codes.ok)
+
+        with self.subTest(f"{replica}: Bad returned when uploading a file with an invalid version"):
+            self.upload_file(source_url, file_uuid, version='', expected_code=requests.codes.bad)
 
-        # should be able to do this twice (i.e., same payload, same UUIDs)
-        self.upload_file(source_url, file_uuid, bundle_uuid=bundle_uuid,
-                         version=version, expected_code=requests.codes.ok)
+        invalid_version = 'ABCD'
+        with self.subTest(f"{replica}: server_error returned "
+                          f"when uploading a file with invalid version {invalid_version}"):
+            self.upload_file(source_url, file_uuid, version=invalid_version, expected_code=requests.codes.server_error)
```

Comment:
The return code is changed from `bad` to `server_error`?

Reply:
there is an issue open for this #1434